deepchem/data/data_loader.py

```diff
@@ -1742,5 +1742,9 @@ def _featurize_shard(self, shard):
             raise ValueError(
                 "Unknown key in yaml file. Please check format for correctness."
             )
```

Comment:
`_get_shards` should be a generator that yields a shard's worth of data at a time. It will need to be fixed in order to implement shard_size > 1 support.

Reply:
As a second note, `DFTYamlLoader` is currently broken for the case of multiple yaml files. Can you add a test to check the case of loading multiple file

Reply:
_featurize_shard will need to be changed to handle shard_size > 1 since it currently only featturizes a single datapoint at a time